### PR TITLE
odyssey tests with asan, little test refactor

### PR DESCRIPTION
**Pull request by reshke**

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -10,4 +10,8 @@ ody-stop
 
 /ody-intergration-test
 
+ody-stop
+
+/usr/bin/odyssey-asan /etc/odyssey/odyssey.conf && /shell-test/test.sh
```
Comment (reshke) [inline docker/entrypoint.sh:15]: if case of some issue exit code should be != 0

Comment (reshke) [reply docker/entrypoint.sh:15]: ci would fail in this case

File: sources/frontend.c [hunk 2]
```diff
@@ -743,17 +743,10 @@ od_frontend_remote(od_client_t *client)
 	od_server_t *server;
 	od_instance_t *instance = client->global->instance;
 	for (;;) {
-		/* we go to the eternal wait
```
Comment (x4m) [inline sources/frontend.c:746]: is this change related to PR topic?

File: sources/id.h [hunk 3]
```diff
@@ -48,7 +49,9 @@ od_id_generate(od_id_t *id, char *prefix)
 		id->id[w++] = hex[(seed[q] >> 4) & 0x0F];
 		id->id[w++] = hex[(seed[q]) & 0x0F];
 	}
+#if OD_DEVEL_LVL != -1
```
Comment (x4m) [inline sources/id.h:52]: Assert if free in release build anyway

Comment (reshke) [reply sources/id.h:52]: compiler warning

**Review by x4m — APPROVED** [review]

File: sources/frontend.c (hunk 2 above)
Comment (reshke) [reply sources/frontend.c:746]: this is completely wrong and should be removed,  test on this behaviour is wrong too
